Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,590 @@
+  if (!title) title = this.dialogueBranchTitle;
+  return (
+    Object.keys(this.runner.visited).includes(title) &&
+    this.runner.visited[title]
+  );
+};
+
+/**
+ * Get the entire unparsed text of the current Dialogue Branch
+ */
+gdjs.dialogueTree.getBranchText = function() {
+  if (this.dialogueIsRunning) {
+    return this.dialogueBranchBody;
+  }
+  return "";
+};
+
+/**
+ * Get the value of a variable that was created by the Dialogue parses.
+ * @param {string} key The name of the variable you want to get the value of
+ */
+gdjs.dialogueTree.getVariable = function(key) {
+  if (this.dialogueIsRunning && key in this.runner.variables.data) {
+    return this.runner.variables.data[key];
+  }
+  return "";
+};
+
+/**
+ * Check if a specific variable created by the Dialogue parses exists and is equal to a specific value.
+ * @param {string} key The name of the variable you want to check the value of
+ * @param {string} value The value you want to check against
+ */
+gdjs.dialogueTree.compareVariable = function(key, value) {
+  if (this.dialogueIsRunning && key in this.runner.variables.data) {
+    return this.runner.variables.data[key].toString() === value;
+  }
+  return false;
+};
+
+/**
+ * Set a specific variable created by the Dialogue parser to a specific value.
+ * @param {string} key The name of the variable you want to set the value of
+ * @param {string} value The value you want to set
+ */
+gdjs.dialogueTree.setVariable = function(key, value) {
+  if (this.dialogueIsRunning && this.runner.variables.data) {
+    this.runner.variables.data[key] = value;
+  }
+};
+
+/**
+ * Store the current State of the Dialogue Parser in a specified variable.
+ * Can be used to implement persistence in dialogue through your game's Load/Save function.
+ * That way you can later load all the dialogue choices the player has made.
+ * @param {gdjs.Variable} storeVar The variable where to store the State
+ */
+gdjs.dialogueTree.saveState = function(storeVar) {
+  const dialogueState = {
+    variables: gdjs.dialogueTree.runner.variables.data,
+    visited: gdjs.dialogueTree.runner.visited
+  };
+  gdjs.evtTools.network._objectToVariable(dialogueState, storeVar);
+};
+
+/**
+ * Load the current State of the Dialogue Parser from a specified variable.
+ * Can be used to implement persistence in dialogue through your game's Load/Save function.
+ * That way you can later load all the dialogue choices the player has made.
+ * @param {gdjs.Variable} storeVar The variable where to load the State from.
+ */
+gdjs.dialogueTree.loadState = function(storeVar) {
+  const jsonData = gdjs.evtTools.network.variableStructureToJSON(storeVar);
+  try {
+    const loadedState = JSON.parse(
+      gdjs.evtTools.network.variableStructureToJSON(storeVar)
+    );
+    gdjs.dialogueTree.runner.visited = loadedState.visited;
+    gdjs.dialogueTree.runner.variables.data = loadedState.variables;
+  } catch (e) {
+    console.error(e);
```

Comment:
Be more precise in the error message, so that it's obvious for someone reading the console that something happened related to this extension:
```suggestion
    console.error("Error while trying to parse Dialogue Tree saved state from a variable:", error);
```

Prefer also full variable name rather than "e". Easier to search in a file.